### PR TITLE
carps-cups: init at unstable-2018-03-05

### PR DESCRIPTION
**Pull request by ewok-old**
###### Motivation for this change
Adding support for old Canon printers(from project page):
Printer | Status
-|- 
MF5730 | works
MF5750 | works
MF5770 | should work
MF5630 | works
MF5650 | should work
MF3110 | works
imageCLASS D300 | works
LASERCLASS 500 | should work
FP-L170/MF350/L380/L398 | should work
LC310/L390/L408S | works
PC-D300/FAX-L400/ICD300 | works
L180/L380S/L398S | works

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ewok** [comment]
I have MF3110 tested it works

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchgit
+, cups
+}:
+
+stdenv.mkDerivation {
+  pname = "carps-cups";
+  version = "18d80d1";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:9]:
```suggestion
  version = "unstable-XXXX-XX-XX";
  
  src = fetchgit {
```

Please use the date of the commit.

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    url = "https://github.com/ondrej-zary/carps-cups.git";
+    rev = "18d80d1d6f473dd9132e4b6d8b5c592c74982f17";
+    sha256 = "0mjj9hs5lqxi0qamgb4sxfz4fvf7ggi66bxd37bkz3fl0g9xff70";
+  };
+
+  buildPhase = ''
+    export CUPS_DATADIR="${cups}/share/cups"
+    make
+  '';
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:18]:
```suggestion
  preBuild = ''
    export CUPS_DATADIR="${cups}/share/cups"
  '';
```

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  installPhase = ''
+    CUPSDIR="$out/lib/cups"
+    CUPSDATADIR="$out/share/cups"
+
+    mkdir -p "$CUPSDIR/filter"
+    mkdir -p "$CUPSDATADIR/drv"
+    mkdir -p "$CUPSDATADIR/usb"
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:26]:
```suggestion
    mkdir -p $CUPSDIR/filter $CUPSDATADIR/drv $CUPSDATADIR/usb
```

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    install -s rastertocarps $CUPSDIR/filter
+    install -m 644 carps.drv $CUPSDATADIR/drv/
+    install -m 644 carps.usb-quirks $CUPSDATADIR/usb/
+  '';
+
+  buildInputs = [ cups ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:35]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    install -m 644 carps.drv $CUPSDATADIR/drv/
+    install -m 644 carps.usb-quirks $CUPSDATADIR/usb/
+  '';
+
+  buildInputs = [ cups ];
+
+  meta = with lib; {
+    description = "CUPS Linux drivers for Canon printers";
+    homepage = "http://www.canon.com/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:38]: gpl3Only or gpl3Plus?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/misc/cups/drivers/carps-cups/default.nix (hunk 5 above)
Comment (ewok) [reply pkgs/misc/cups/drivers/carps-cups/default.nix]: gpl3Plus. Updated, thank you!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>carps-cups</li>
</ul>
</details>

File: pkgs/misc/cups/drivers/carps-cups/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    install -s rastertocarps $CUPSDIR/filter
+    install -m 644 carps.drv $CUPSDATADIR/drv/
+    install -m 644 carps.usb-quirks $CUPSDATADIR/usb/
+  '';
+
+  buildInputs = [ cups ];
+
+  meta = with lib; {
+    description = "CUPS Linux drivers for Canon printers";
+    homepage = "http://www.canon.com/";
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/carps-cups/default.nix:37]:
```suggestion
    homepage = "https://www.canon.com/";
```

**Pull request merged**